ledger/src/blockstore_meta.rs

```diff
@@ -365,6 +412,51 @@ pub struct DuplicateSlotProof {
     pub shred2: shred::Payload,
 }
 
+#[derive(Deserialize, Serialize)]
+pub enum BlockStatus {
+    /// The block is being ingested, of the shreds received there are no conflicts
+    Incomplete,
+
+    /// The block `block_id` has been fully ingested, has consistent shreds and is ready for replay
+    Complete { block_id: Hash },
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct BlockVersions {
+    pub turbine_version: BlockStatus,
+    pub repaired_versions: [(Hash, BlockStatus); 3],
```

Comment:
Tuples can be hard to track, would it be cleaner to augment the BlockStatus::Incomplete to also track the first hash it gets?

Can then implement a `BlockStatus::hash() -> Hash` function for convenience

Reply:
I don't think that works, by definition we can only know the block hash once it is complete

Reply:
Hmm, so what is stored in the Hash of this tuple before we receive enough of the block, just Hash::default?

Reply:
It's whatever is tagged onto the repair request. I might have been misunderstanding your initial question, were you suggesting that Incomplete also uses the hash from the repair request? Because then the Turbine column would have to use Hash::default or some different sentinel